kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp

```diff
@@ -151,6 +151,16 @@ pcl::KdTreeFLANN<PointT, Dist>::nearestKSearch (const PointT &point, int k,
                            k_indices_mat, k_distances_mat,
                            k, param_k_);
 
+  // Filter out invalid indices, which can occur when infinite or NaN values are in the cloud
+  const size_t& max_index = identity_mapping_ ? total_nr_points_ : index_mapping_.size();
```

Comment:
index_t here and use auto in L157, that'd remove the need for static_cast in L158

Reply:
Done